spec/v1.0/statement.md

```diff
@@ -33,37 +33,24 @@ Additional [parsing rules] apply.
 > Identifier for the schema of the Statement. Always
 > `https://in-toto.io/Statement/v1` for this version of the spec.
 
-`subject` _array of objects, required_
+`subject` _array of [ResourceDescriptor] objects, required_
 
 > Set of software artifacts that the attestation applies to. Each element
-> represents a single software artifact.
+> represents a single software artifact. Each object MUST have `digest` set.
+> The `name` field may be used as an identifier to distinguish this artifact
+> from others within the `subject`. The semantics are up to the producer and
+> consumer and they MAY use it when evaluating policy. If the name is not
+> meaningful, leave the field unset or use "\_". For example, a
+> [SLSA Provenance] attestation might use the name to specify output filename,
+> expecting the consumer to only consider entries with a particular name.
+> Alternatively, a vulnerability scan attestation might leave name unset because
+> the results apply regardless of what the artifact is named. If set, `name`
+> SHOULD be unique within subject.
```

Comment:
Is there a reason the uniqueness of `name` is only a SHOULD?

Reply:
Because we don't have a reason to make be a MUST 😜 . What is the rationale for making the file invalid if two names are the same?

/cc @TomHennen

Reply:
I was thinking about it from the perspective of policies that rely on the `name` field. Taken with #211, consider the link predicate. The subject field is essentially the products field in the old scheme. To apply in-toto's artifact rules, we rely on `name` before using the `digests`. In the original schema, the verifier is only going to get one entry for a `name` as it uses a map but in the new model, we'd have to separately enforce uniqueness of `name`.

I think this ties into a broader question. Can predicates set expectations on the `Statement` wrapping them? If a link predicate can mandate that `digest` is compulsory for its materials field (within the predicate), can it do the same for unique `name` for the subjects?

Reply:
>  Can predicates set expectations on the Statement wrapping them?

This seems useful, and a good compromise?

Reply:
I think it could work indeed.

Validation with something like https://www.kusari.dev/blog/spector-announcement/ could get more complicated, though. Especially for conditions like unique `name` across all subjects rather than just requiring some field in every subject entry. cc @mlieberman85, @pxp928

Reply:
I'm going to resolve this, I think the current consensus is to let predicates specify required fields. It's probably a breaking change to require `name` in the spec? :thinking: